easybuild/framework/easyconfig/types.py

```diff
@@ -39,14 +39,16 @@
 # specific type: dict with only name/version as keys, and with string values
 # additional type requirements are specified as tuple of tuples rather than a dict, since this needs to be hashable
 NAME_VERSION_DICT = (dict, (('only_keys', ('name', 'version')), ('value_types', (str,))))
+DEPENDENCIES = (list, (('element_types', (tuple,)),))
```

Comment:
`dependencies` are converted to `dict` values, no? which have a set of allowed keys (name/version/toolchain/versionsuffix)?